src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,166 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 5;
+    private Object[] defaultNewCapacity;
+    private T[] elementArray;
+    private int size = 0;
+
+    public ArrayList() {
+        elementArray = (T[]) new Object[DEFAULT_SIZE];
+    }
+
+    public ArrayList(int capacity) {
+        elementArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementArray.length) {
+            regenerationSizeArray();
+        }
+        elementArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(" Index " + index
+                    + " out of bounds for length " + size);
+        }
+        defaultNewCapacity = new Object[size + 1];
+        for (int i = 0; i < defaultNewCapacity.length; i++) {
+            if (i < index) {
+                defaultNewCapacity[i] = elementArray[i];
+            }
+            if (i == index) {
+                defaultNewCapacity[i] = value;
+            }
+            if (i > index) {
+                defaultNewCapacity[i] = elementArray[i - 1];
+            }
+        }
+        size++;
+        elementArray = (T[]) defaultNewCapacity;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() > size || list.size() < 0) {
+            throw new ArrayListIndexOutOfBoundsException("list " + list.size()
+                    + " out of bounds for length " + size);
+        }
+        for (int i = 0; i < list.size(); i++) {
+            elementArray[size + i] = list.get(i);
+        }
+        size = size + list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + elementArray.length);
+        }
+        return elementArray[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " out of bounds for length " + size);
+        }
```

Comment:
[Common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#dont-create-repeating-code)
**Don’t create repeating code.**
If the logic of your code repeats - move it to the separate private method. For example, the index check for get(), set() and remove() methods should be moved to the separate method.